### DISCUSSION TITLE
Testing with jest or any other library

### DISCUSSION THREAD

**Discussion by yann-lauwers** [dinerojs/dinero.js#747] [Q&A] [answered]
Hi everyone,

I am looking for the best way to run tests on the utils I created with Dinero.

I've tried `expect(myCustomDineroUtils().toJSON()).toEqual(expectedAmount.toJSON());` and it seems to work fine.

_myCustomDineroUtils returns a Dinero.Dinero object_

Is there a better solution has anyone some tips to test my utils ? 

Thank you. Looking forward for you help!

**Comment by sarahdayan** [comment, marked as answer]
Hey @yann-lauwers! The recommended way to compare Dinero objects in tests is to use the [`equal`](https://v2.dinerojs.com/api/comparisons/equal.html) function:

```js
import { dinero, equal } from 'dinero.js';
import { USD } from 'dinero.js/currencies';

expect(equal(result, dinero({ amount: 5000, currency: USD }))).toBe(true);
```

If you need to inspect the raw values (e.g. for snapshot testing), [`toSnapshot`](https://v2.dinerojs.com/api/utilities/to-snapshot.html) returns a plain object you can assert on:

```js
import { toSnapshot } from 'dinero.js';

expect(toSnapshot(result)).toEqual({ amount: 5000, currency: USD, scale: 2 });
```
